src/improvements/tasks/MultisigTask.sol

```diff
@@ -893,7 +921,9 @@ abstract contract MultisigTask is Test, Script, ITask {
     /// check the state changes applied by the task. This function can check
     /// that only the nonce changed in the parent multisig when executing a task
     /// by checking the slot and address where the slot changed.
-    function checkStateDiff(VmSafe.AccountAccess[] memory) internal view virtual {}
+    function checkStateDiff(VmSafe.AccountAccess[] memory) internal view virtual {
+        revert("checkStateDiff must be implemented");
```

Comment:
Maybe we are still miscommunicating here. The default implementation of this method should be identical to what it is in the legacy superchain-ops, and it should run on every task (every template) by default. This PR requires every task template to reimplement this method https://github.com/ethereum-optimism/superchain-ops/blob/9f1cb1aa2f8bf3938289a50e2ff5317dd22c773f/src/JsonTxBuilderBase.sol#L59

The new `validate` signature and where we call `checkStateDiff` LGTM now, it's just this default implementation that we need to fix. I will push a commit to this PR so it's clear what I mean

Reply:
Thank you! Let's make sure to add tests for this new functionality too